src/modules/validator.js

```diff
@@ -0,0 +1,96 @@
+class Validator {
+
+    constructor(chluIpfs) {
+        this.chluIpfs = chluIpfs;
+    }
+
+    async validateReviewRecord(reviewRecord, validations = {}) {
+        const rr = cloneReviewRecord(reviewRecord);
+        const v = Object.assign({
+            validateVersion: true,
+            validateMultihash: true,
+            validateHistory: true
+        }, validations);
+        if (v.validateVersion) this.validateVersion(rr);
+        if (v.validateMultihash) await this.validateMultihash(rr, rr.hash.slice(0));
+        if (v.validateHistory) await this.validateHistory(rr);
+    }
+
+    async validateMultihash(obj, expected) {
+        const hashedObj = await this.chluIpfs.reviewRecords.hashReviewRecord(obj);
+        if (expected !== hashedObj.hash) {
+            throw new Error('Mismatching hash');
+        }
+    }
+
+    async validateHistory(reviewRecord) {
+        const history = await this.chluIpfs.reviewRecords.getHistory(reviewRecord);
+        if (history.length > 0) {
+            const opt = {
+                validate: false
+            };
+            const rr = await this.chluIpfs.reviewRecords.readReviewRecord(history[0], opt);
+            this.validatePrevious(reviewRecord, rr);
```

Comment:
`validatePrevious` is essentially validating the popr, why not call it `validatePoPRIsUnchanged`.

Reply:
It also checks other fields not part of the PoPR so I don't think it would be correct to change the name. The point of the function is to check that even though it got updated the reviewRecord didn't have amounts and addresses changed